src/include/clSPARSE.h

```diff
@@ -360,6 +360,40 @@ clsparseDCsrMatrixfromFile( clsparseCsrMatrix* csrMatx, const char* filePath, cl
         const cldenseVector* y,
         const clsparseControl control );
 
+    // functions for tests of scan transformation, as simple as possible
```

Comment:
I'm curious, why do you think we should provide scan and reduce API's in clSPARSE; is this common for sparse operation?  I hesitate to pollute the API with tertiary functionality.  If a user requires this functionality, shouldn't they use Bolt or other library dedicated to this functionality?  

The point I want to make, for every API we expose, we increase our support and maintenance costs.  What if in the future we eliminate our internal needs for scan or reduce?  Does clSPARSE then still have the burden of providing this functionality to end users?

Our default mentality should be to eliminate API's, and support a minimal set.  Justification needs to be provided before new API's are added, because this increases our surface area for support.